.github/workflows/deploy-preview.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy PR Preview
+on:
+  pull_request:
+    branches: [main]
+    types:
+      - opened
+      - reopened
+      - synchronize
+permissions:
+  contents: read
+  pages: write
+  pull-requests: write
+  id-token: write
+concurrency: preview-${{ github.ref }}
+jobs:
+  deploy-preview:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+      - name: Use Node.js 18.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+      - name: Install and Build 🔧
+        run: |
+          npm ci --include=dev
+          npm run build
+          npm run style
+          cp src/configs/config.aws.js src/config.js
+      - name: Remove prior FTP preview
+        uses: StephanThierry/ftp-delete-action@v2.1
+        with:
+          host: ${{ secrets.FTP_SERVER }}
+          user: ${{ secrets.FTP_USERNAME }}
+          password: ${{ secrets.FTP_PASSWORD }}
+          remoteDirectories: pr/${{ github.event.pull_request.number }}/
+      - name: Deploy via FTP
+        uses: SamKirkland/FTP-Deploy-Action@v4.3.4
+        with:
+          server: ${{ secrets.FTP_SERVER }}
+          username: ${{ secrets.FTP_USERNAME }}
+          password: ${{ secrets.FTP_PASSWORD }}
+          local-dir: dist/
+          server-dir: pr/${{ github.event.pull_request.number }}/
+          dangerous-clean-slate: true
+      - name: Comment Pull Request
+        uses: thollander/actions-comment-pull-request@v2.3.1
+        with:
+          message: |
+            PR Preview:
+            * [Map](https://preview.ourmap.us/pr/${{ github.event.pull_request.number }}/)
+            * [Shield Test](https://preview.ourmap.us/pr/${{ github.event.pull_request.number }}/shieldtest.html)
+            * [style.json](https://preview.ourmap.us/pr/${{ github.event.pull_request.number }}/style.json)
+            * [taginfo.json](https://preview.ourmap.us/pr/${{ github.event.pull_request.number }}/taginfo.json)
+
+            Sprite Sheets:
+
+            ![Sprites 1x](https://preview.ourmap.us/pr/${{ github.event.pull_request.number }}/sprites/sprite.png)
+            ![Sprites 2x](https://preview.ourmap.us/pr/${{ github.event.pull_request.number }}/sprites/sprite@2x.png)
```

Comment:
When using a 2× display, I’d find it more useful to view the 2× spritesheet at the same width as the 1× spritesheet. Otherwise, this just tells me at a glance that the icons can be twice as large in a spritesheet twice as large.

GitHub supports HTML `<img>` syntax but lamentably sanitizes the `srcset` attribute, so we’d have to figure out a `width` to specify. Is it possible to know the width ahead of time?

Reply:
We could pick a width, but it'd be wrong on some set of displays. On mobile they are the same width.

Reply:
Could we use a width percentage somehow?

Reply:
I tried some various CSS hacks to get this to happen and failed.

Reply:
GitHub recently [added support for “responsive” images](https://github.blog/changelog/2022-08-15-specify-theme-context-for-images-in-markdown-ga/), but only based on light mode versus dark mode. If I try something like:

```html
<picture>
<source srcset="https://preview.ourmap.us/pr/868/sprites/sprite@2x.png" media="(min-resolution: 2dppx)">
<img src="https://preview.ourmap.us/pr/868/sprites/sprite.png" alt="">
</picture>
```

the `<source>` has no effect. There appears to be some Web Components magic going on that overrides the image with custom image switching that doesn’t understand resolutions.

It’s OK, not a blocker. I just open the image in a new tab and zoom out until it looks the same size.

Reply:
Feel free to edit the auto comment to experiment.

Reply:
Yeah, I threw everything I had at it, but HTML that should produce the right result does not, apparently because of some Web Components–based polyfill in JavaScript that overrides the correct image with the incorrect image based on the theme.

Reply:
Resolving conversation as CANTFIX :grinning: